Assets/Scripts/NPCs/PureChaser.cs

```diff
@@ -1,105 +1,111 @@
-﻿using System.Collections;
-using System.Collections.Generic;
+﻿using NPCs.Components;
 using UnityEngine;
-using UnityEngine.AI;
 
-public class PureChaser : MonoBehaviour
+namespace NPCs
 {
-    public Transform targetTransform;
-    public float startChaseRadius = 100f;
-
-    private ChildRootMotionController rootMotionController;
-    private NavMeshAgent agent;
-    private bool shouldChase = false;
-
-    public const float reportReachedDistance = 2f;
-
-    public event Chaser.CollideWithPlayerAction OnCollideWithPlayer;
-    public delegate void ReachedDestinationAction();
-    public event ReachedDestinationAction OnReachDestination;
-
-    private void Awake()
+    [RequireComponent(typeof(Chaser))]
+    public class PureChaser : BaseAi
     {
-        agent = Utils.GetRequiredComponent<NavMeshAgent>(this);
-        rootMotionController = Utils.GetRequiredComponentInChildren<ChildRootMotionController>(this);
-    }
+        public float startChaseRadius = 100f;
+        
+        [HideInInspector] public Chaser chaser;
+        private ChildRootMotionController rootMotionController;
+        private bool shouldChase = false;
 
-    public void SetSpeed(float speed)
-    {
-        agent.speed = speed;
-    }
+        public const float ReportReachedDistance = 2f;
+        
 
-    public void SetAnimationSpeed(float speed)
-    {
-        rootMotionController.SetAnimationSpeed(speed);
-    }
+        public delegate void ReachedDestinationAction();
+        public event ReachedDestinationAction OnReachDestination;
 
-    public void SetDestination(Vector3 position)
-    {
-        agent.SetDestination(position);
-    }
-
-    private void ChaseTarget()
-    {
-        if (shouldChase && targetTransform)
+        private new void Awake()
```

Comment:
Wait why do we need this to be a `new` function? Can we just make it override (and make the Awake function in BaseAI virtual)? Same for the Nurse script as well. From what I know, `new` is like hiding the base implementation all together and not commonly used usually... but I don't really know too much about this topic